src/Hid.Net/Windows/WindowsHidApiService.cs

```diff
@@ -139,7 +139,7 @@ public Guid GetHidGuid()
 
         public Stream OpenRead(SafeFileHandle readSafeFileHandle, ushort readBufferSize) => new FileStream(readSafeFileHandle, FileAccess.Read, readBufferSize, true);
 
-        public Stream OpenWrite(SafeFileHandle writeSafeFileHandle, ushort writeBufferSize) => new FileStream(writeSafeFileHandle, FileAccess.ReadWrite, writeBufferSize, true);
+        public Stream OpenWrite(SafeFileHandle writeSafeFileHandle, ushort writeBufferSize) => writeBufferSize > 0 ? new FileStream(writeSafeFileHandle, FileAccess.ReadWrite, writeBufferSize, true) : Stream.Null;
```

Comment:
@brendonparker I don't think we should ever return a null stream. IF the write buffer size is invalid, it should throw an exception. I would say that the bug must be elsewhere.

Reply:
Would you like me to write a new issue, or just re-phrase the current issue in this issue thread? Happy to do either.